File: 0123/123.ts
function bestTimeToBuyAndSellStockIii(root: TreeNode | null): number {
    if (!root) return 0;
    return 1 + Math.max(bestTimeToBuyAndSellStockIii(root.left), bestTimeToBuyAndSellStockIii(root.right));
}

File: 0123/123.go
func bestTimeToBuyAndSellStockIii(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := bestTimeToBuyAndSellStockIii(root.Left)
	r := bestTimeToBuyAndSellStockIii(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}